Automattic/jetpack
# Licensing: Add GoldenTokenModal on manual enter of key

Makes the GoldenToken experience happen when manually applying a license. 

The GoldenTokenModal component can also represent different states, depending on whether the license is already redeemed. 
- If it's not redeemed, the existing behavior is present. It basically shows a button that doesn't do anything. We could add this in a later PR, or brute force a text field that makes the same assigning API calls. 
- If it is redeemed, it congratulates them. The experience is mostly the same on the next page. 
- It adds some default animation that happens on load of the modal. For extra pizzaz. 

## Proposed changes:

- moves the GoldenTokenModal to the licensing package, so that it can be rendered on manual entering of a token license. 
- My Jetpack now consumes this component from the licensing package.
- updates the component to be more self-reliant.
- adds `classname` as a dependency to the licensing js package, so that it can do the same css class hashing that was done before. 

### Other information:
- The MC tooling can actually give a license. 
See it in action: p1680014276014749-slack-C04V237FP40

## Jetpack product discussion



## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






- Connect Jetpack fully.

Apply the license manually to existing UI in wp-admin 
- Generate a new golden-token license key in the MC tool. 
- Go to `/wp-admin/admin.php?page=my-jetpack#/add-license`, and enter the key manually
- Check out the experience. 

Assign the license from the MC tool
- This flow mimics a flow in which we choose to manually provision the license for the site. 
- So, use the MC tool to assign it to your test site. 
- Then go to the special route `/wp-admin/admin.php?page=my-jetpack#/redeem-token` to make sure the component renders from My Jetpack properly. 

If you need to know how to revoke the license to test again, you can do so using the jetpack-start-gui MC tool.
